lib/jet_filters/operator/comparable.ex

```diff
@@ -0,0 +1,55 @@
+defmodule JetFilters.Operator.GT do
+  @moduledoc false
+
+  use JetFilters.Operator, [
+    [:string, :string],
+    [:number, :number],
+    [:datetime, :datetime]
+  ]
+
+  build_to_dynamic([_type1, _type2]) do
+    dynamic(^op1 > ^op2)
+  end
```

Comment:
这样呢？放到编译时来做，是不是就不需要运行时使用 determine_operand_types 来检查了？
```suggestion
  operand_types = [
    [:string, :string],
    [:number, :number],
    [:datetime, :datetime]
  ]
  use JetFilters.Operator, operand_types

  for [type, type] <- operand_types do
    build_to_dynamic([type, type]) do
      dynamic(^op1 > ^op2)
    end
  end
```

Reply:
我现在的写法运行时其实就不会校验这个东西。`determine_operand_types` 是在更早的一个阶段完成的（校验阶段），到了 `to_dynamic` 这里的时候，来的 operands 一定是合法的了。这里还要再传一次 operand_types 的原因是要做多态，比如 `Operator.CO` 对于 `[:string, :string]` 和 `[:array, :array]` 的行为是不一样的

Reply:
1. 编译时做到 function head 上面去，应该可以省略 determine_operand_types，那里会有模式匹配判定是否是支持的
2. `use JetFilters.Operator` 的参数和 `build_to_dynamic` 可以合并到一起，现在相当于写了两次这个参数

Reply:
2. 这里的参数和上面的参数用途是不一样的，上面是为了校验，下面是为了多态，你看不需要多态的地方我都是 `[_type1, _type2]`，我是不关心它的类型的

Reply:
1. 校验转移到 1 中了（function head）
2. 多态中的 type1 和 type2 的类型是不是明确的，或者是 pre-runtime 的设计？

Reply:
2. operands 进入到 to_dynamic 这个函数的时候，是已经过了校验的，一定是合法的；to_dynamic 的参数还要传 operand_type，是为了多态，例如上面说的 Operator.CO；对于不需要多态的，此时就不关心这个类型，所以就用下划线参数

Reply:
新的实现里面的 type_matcher 和 to_dynamic clauses 是一一对应吗？
我想的是我们再用 macro 的时候，就不分成两部分去写了

Reply:
不一样的，你再看一下，就像我上面说的，只有少数地方需要多态，才在 to_dynamic 的参数上做的模式匹配